src/odemis/driver/test/npmc_test.py

```diff
@@ -102,61 +109,104 @@ def setUpClass(cls):
-        shift = {'x': 0.12}
-        f = self.dev.moveRel(shift)
-        f.result()
-        test.assert_pos_almost_equal(add_coord(old_pos, shift), self.dev.position.value)
+        for shift in (0, 0.12, -0.12):
+            new_shift = {'x': shift}
+            old_pos = self.dev.position.value
+            f = self.dev.moveRel(new_shift)
+            f.result()
+            test.assert_pos_almost_equal(add_coord(old_pos, new_shift), self.dev.position.value, **COMP_ARGS)
 
     def test_offset(self):
+        # Start at the origin to prevent hitting the ends
+        self.dev.moveAbs({'x': 0}).result()
+
         # Bad offset
         with self.assertRaises(ValueError):
             self.dev.updateMetadata({model.MD_POS_COR: 5})
 
         # Test offset #1
         old_pos = self.dev.position.value
-        offset_1 = {'x': 0.1}
+        offset_1 = {'x': 0.05}
         self.dev.updateMetadata({model.MD_POS_COR: offset_1})
-        test.assert_pos_almost_equal(subtract_coord(old_pos, offset_1), self.dev.position.value)
+        test.assert_pos_almost_equal(subtract_coord(old_pos, offset_1), self.dev.position.value, **COMP_ARGS)
 
         # Test offset #2
         old_pos = self.dev.GetRealPosition()
-        offset_2 = {'x':-1}
+        offset_2 = {'x':-0.05}
         self.dev.updateMetadata({model.MD_POS_COR: offset_2})
-        test.assert_pos_almost_equal(subtract_coord(old_pos, offset_2), self.dev.position.value)
+        test.assert_pos_almost_equal(subtract_coord(old_pos, offset_2), self.dev.position.value, **COMP_ARGS)
+
         # now move to the origin
         new_pos = {'x': 0}
         f = self.dev.moveAbs(new_pos)
         f.result()
-        test.assert_pos_almost_equal(offset_2, self.dev.GetRealPosition())
+        test.assert_pos_almost_equal(offset_2, self.dev.GetRealPosition(), **COMP_ARGS)
+
+        # Remove the offset
+        self.dev.updateMetadata({model.MD_POS_COR: {'x': 0}})
+
+    def test_error_handling(self):
+        """
+        Test the error handling when it hits a limit
+        """
+        with self.assertRaises(npmc.ESPError):
+            self.dev.moveAbs({'x': 1}).result()
+        self.dev.moveAbs({'x': 0}).result()
+
+    def test_cancellation(self):
+        """
+        Test queuing up moves and cancelling
+        """
+        # start at origin
+        self.dev.moveAbs({'x': 0}).result()
+
+        for pos in (0.15, -0.15, 0.2, -0.1):
+            new_pos = {'x': pos}
+            f = self.dev.moveAbs({'x': 0.15})
+            time.sleep(0.1)
+            f.cancel()
+
+            # make sure the posiiton is not the new position
+            difference = new_pos['x'] - self.dev.position.value['x']
+            self.assertNotEqual(round(difference, 4), 0)
+
+    def test_multimove(self):
+        """
+        Test running multimoves
+        """
+
+        # start at origin
+        self.dev.moveAbs({'x': 0}).result()
+
+        f1 = self.dev.moveRel({'x': 0.15})
+        time.sleep(0.1)
```

Comment:
Depending on the duration of the move, 0.1 s might be quite long. You could just set it to 0.02s, to be more certain that the first move hasn't finished by the time you're queuing the second one.